src/common/mfu_daos.c

```diff
@@ -1313,3 +1330,2195 @@ int mfu_flist_copy_daos(
+    if (hdf5->oid_memtype < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create oid memtype");
+        rc = 1;
+        goto out;
+    }
+    hdf5->status = H5Tinsert(hdf5->oid_memtype, "OID Hi",
+                             HOFFSET(oid_t, oid_hi), H5T_NATIVE_UINT64);
+    if (hdf5->status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to insert oid hi");
+        rc = 1;
+        goto out;
+    }
+    hdf5->status = H5Tinsert(hdf5->oid_memtype, "OID Low",
+                             HOFFSET(oid_t, oid_low), H5T_NATIVE_UINT64);
+    if (hdf5->status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to insert oid low");
+        rc = 1;
+        goto out;
+    }
+    hdf5->status = H5Tinsert(hdf5->oid_memtype, "Dkey Offset",
+                             HOFFSET(oid_t, dkey_offset), H5T_NATIVE_UINT64);
+    if (hdf5->status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to insert dkey offset");
+        rc = 1;
+        goto out;
+    }
+    hdf5->dkey_memtype = H5Tcreate(H5T_COMPOUND, sizeof(dkey_t));
+    if (hdf5->dkey_memtype < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create dkey memtype");
+        rc = 1;
+        goto out;
+    }
+    hdf5->dkey_vtype = H5Tvlen_create(H5T_NATIVE_CHAR);
+    if (hdf5->dkey_vtype < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create dkey vtype");
+        rc = 1;
+        goto out;
+    }
+    hdf5->status = H5Tinsert(hdf5->dkey_memtype, "Akey Offset",
+                             HOFFSET(dkey_t, akey_offset),
+                             H5T_NATIVE_UINT64);
+    if (hdf5->status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to insert akey offset");
+        rc = 1;
+        goto out;
+    }
+    hdf5->status = H5Tinsert(hdf5->dkey_memtype, "Dkey Value",
+                             HOFFSET(dkey_t, dkey_val), hdf5->dkey_vtype);
+    if (hdf5->status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to insert dkey value");
+        rc = 1;
+        goto out;
+    }
+    hdf5->akey_memtype = H5Tcreate(H5T_COMPOUND, sizeof(akey_t));
+    if (hdf5->akey_memtype < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create akey memtype");
+        rc = 1;
+        goto out;
+    }
+    hdf5->akey_vtype = H5Tvlen_create(H5T_NATIVE_CHAR);
+    if (hdf5->akey_vtype < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to create akey vtype");
+        rc = 1;
+        goto out;
+    }
+    hdf5->status = H5Tinsert(hdf5->akey_memtype, "Dataset ID",
+                             HOFFSET(akey_t, rec_dset_id),
+                             H5T_NATIVE_UINT64);
+    if (hdf5->status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to insert record dset id");
+        rc = 1;
+        goto out;
+    }
+    hdf5->status = H5Tinsert(hdf5->akey_memtype, "Akey Value",
+                             HOFFSET(akey_t, akey_val), hdf5->akey_vtype);
+    if (hdf5->status < 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to insert akey value");
+        rc = 1;
+        goto out;
+    }
+out:
+    return rc;
```

Comment:
@dsikich In general, if rc is not 0, do we need to close or cleanup anything here?